src/vcpkg/portfileprovider.cpp

```diff
@@ -307,18 +309,21 @@ namespace vcpkg
                             {
                                 return SourceControlFileAndLocation{std::move(scf), std::move(ports_spec)};
                             }
-                            Checks::exit_maybe_upgrade(
+                            Checks::msg_exit_maybe_upgrade(
                                 VCPKG_LINE_INFO,
-                                "Error: Failed to load port from %s: names did not match: '%s' != '%s'",
-                                ports_spec,
-                                port_name,
-                                scf->core_paragraph->name);
+                                msg::format(msgFailedToLoadPort, msg::package_name = port_name, msg::path = ports_spec)
+                                    .append_raw("\n")
+                                    .append(msgMismatchedNames,
+                                            msg::package_name = port_name,
+                                            msg::actual = scf->core_paragraph->name));
```

Comment:
I think these should be combined into one message; generally we shouldn't assume that localized messages can be combined unless they are clearly independent messages. (For example, they may need to be combined in a different order in different languages)

Reply:
I believe these messages are independent.